service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java

```diff
@@ -341,13 +340,21 @@ private void getWorkspaceByUserFacingIdExpectingError(
     mockMvc.perform(addAuth(requestBuilder, userRequest)).andExpect(status().is(statusCode));
   }
 
-  private List<ApiWorkspaceDescription> listWorkspaces(AuthenticatedUserRequest request)
-      throws Exception {
-    return listWorkspaces(request, /*minimumHighestRole=*/ Optional.empty());
+  private Optional<ApiWorkspaceDescription> getWorkspaceUsingListWorkspaces(
+      AuthenticatedUserRequest request, UUID workspaceId) throws Exception {
+    return getWorkspaceUsingListWorkspaces(
+        request, workspaceId, /*minimumHighestRole=*/ Optional.empty());
   }
 
-  private List<ApiWorkspaceDescription> listWorkspaces(
-      AuthenticatedUserRequest request, Optional<ApiIamRole> minimumHighestRole) throws Exception {
+  /**
+   * Call listWorkspaces and return workspace with specified workspaceId.
+   *
+   * <p>This should be used instead of just calling listWorkspaces. Tests run in parallel.
+   * listWorkspaces will return workspaces created by a different test on a different thread.
```

Comment:
With the current setup, this shouldn't happen unless you're running with a single DB locally. normally, each thread is using a separate postgres DB container. it's definitely possible a different test on the same thread didn't clean up a workspace, though

Reply:
I see, good to know, thanks! I looked at build scan more closely. Turns out this is because I forgot to delete a workspace in #906, as [Dan pointed out](http s://github.com/DataBiosphere/terra-workspace-manager/pull/906#discussion_r1024269910). Deleting the workspace should make this problem go away. Closing PR.